src/coins.cpp

```diff
@@ -1408,7 +1430,41 @@ CValidationState::Code CCoinsViewCache::IsScTxApplicableToState(const CTransacti
          */
         if (!GetSidechain(scId, sidechain))
         {
-            LogPrintf("%s():%d - ERROR: mbtr not supported\n",  __func__, __LINE__);
+            LogPrintf("%s():%d - ERROR: tx[%s] MBTR output [%s] refers to unknown scId[%s]\n",
+                __func__, __LINE__, tx.ToString(), mbtr.ToString(), scId.ToString());
+            return CValidationState::Code::INVALID;
+        }
+
+        /**
+         * Check that the size of the Request Data field element is the same specified
+         * during sidechain creation.
+         */
+        if (mbtr.vScRequestData.size() != sidechain.fixedParams.mainchainBackwardTransferRequestDataLength)
+        {
+            LogPrintf("%s():%d - ERROR: Invalid tx[%s] : MBTR request data size [%d] must be equal to the size specified "
+                         "during sidechain creation [%d] for scId[%s]",
+                    __func__, __LINE__, txHash.ToString(), mbtr.vScRequestData.size(), sidechain.fixedParams.mainchainBackwardTransferRequestDataLength, scId.ToString());
+            return CValidationState::Code::INVALID;
+        }
+
+        /**
+         * Check that MBTRs are allowed (i.e. the MBTR data length is greater than zero).
+         */
+        if (sidechain.fixedParams.mainchainBackwardTransferRequestDataLength == 0)
+        {
+            LogPrintf("%s():%d - ERROR: mbtr is not allowed for scId[%s]\n",  __func__, __LINE__, scId.ToString());
+            return CValidationState::Code::INVALID;
+        }
+
+        /**
+         * Check that the Mainchain Backward Transfer Request amount is greater than or equal to the
+         * Sidechain Mainchain Backward Transfer Request fee.
+         */
+        if (!CheckScMbtrFee(mbtr))
+        {
+            LogPrintf("%s():%d - ERROR: Invalid tx[%s] : MBTR fee [%s] cannot be less than SC MBTR fee [%s] for scId[%s]",
+                    __func__, __LINE__, txHash.ToString(), FormatMoney(mbtr.scFee),
+                    FormatMoney(GetActiveCertView(scId).mainchainBackwardTransferRequestScFee), scId.ToString());
             return CValidationState::Code::INVALID;
         }
 
```

Comment:
ban in case of if(!sidechain.fixedParams.wCeasedVk.is_initialized())

Reply:
Done